src/main/java/uk/gov/hmcts/reform/divorce/orchestration/tasks/FetchPrintDocsFromDmStoreTask.java

```diff
@@ -111,6 +125,31 @@ private void populateDocumentBytes(TaskContext context, Map<String, GeneratedDoc
         return generatedDocumentInfoList;
     }
 
+    @SuppressWarnings("unchecked")
+    private Map<String, GeneratedDocumentInfo> fromCollectionMember(Map<String, Object> caseData) {
+        List<CollectionMember<Document>> documentList = ofNullable(caseData.get(D8DOCUMENTS_GENERATED))
+            .map(i -> (List<CollectionMember<Document>>) i)
+            .orElse(new ArrayList<>());
+        Map<String, GeneratedDocumentInfo> generatedDocumentInfoList = new HashMap<>();
+
+        for (CollectionMember<Document> document : documentList) {
+            Document value = document.getValue();
+            String documentType = value.getDocumentType();
+            DocumentLink documentLink = ofNullable(value.getDocumentLink()).orElse(null);
+
+            if (ofNullable(documentLink).isPresent()) {
```

Comment:
in the above line you are creating an Optional  and finalising it with else(null) and here you are making it again as optional. You can combine both and use only one optional
```suggestion
            if (ofNullable(value.getDocumentLink()).isPresent()) {
```

Reply:
or just  use value.getDocumentLink() != null

Reply:
done